common/persistence/pinotVisibilityTripleManager.go

```diff
@@ -344,6 +344,18 @@ func (v *pinotVisibilityTripleManager) logUserQueryParameters(userParam userPara
 		tag.VisibilityQuery(userParam.customQuery))
 }
 
+func (v *pinotVisibilityTripleManager) chooseManagerByOverride(ctx context.Context, domain string) VisibilityManager {
+	if override := ctx.Value(ContextKey); override == Primary {
+		v.logger.Debug("primary was chosen")
+		return v.esVisibilityManager
```

Comment:
It is better we also add a nil check, same as the chooseVisibilityManagerForRead()

Reply:
Merged the 2 choose manager functions together. If there are no contextkey, it will fall through into the original logic.

Reply:
Yeah, just in case, since we have nil check guard in the original logic, but no for our override. But this could be fine for now, since no customer will use